lib/fail.js

```diff
@@ -12,79 +13,91 @@ module.exports = async (pluginConfig, context) => {
 
 	logger.log('Sending slack notification on fail')
 
-	const plural = errors.length > 1
+	let slackMessage = {}
+	const repoPath =
```

Comment:
I do not fully understand what this is doing. Could you explain this?

Reply:
It is basically a combination of the previous lines [35](https://github.com/juliuscc/semantic-release-slack-bot/pull/3/files/c075e916d6b7c5cf24ca4adbd6dcedaea7fbb719#diff-997a3fe37340f5a55a3b4ec7f2bad740L35) and [36](https://github.com/juliuscc/semantic-release-slack-bot/pull/3/files/c075e916d6b7c5cf24ca4adbd6dcedaea7fbb719#diff-997a3fe37340f5a55a3b4ec7f2bad740L36) so that repoPath is available for the templated version.

Reply:
Hmm maybe that should not be there? It seems as if it would only work if the repo type is github.

Reply:
Hmm. So the original `if` statement, am I right in thinking that was just adding a message block if GitHub was being used, otherwise say GitLab or BitBucket would still work, they just wouldn't have the message block link?

I was just going of the original code, and was under the impression it was only supposed to work with GitHub.

Reply:
Maybe there should be a separate function which formats the URL so it is clickable, i.e. replaces git@github.com, then other variations can be added as and when. Then we just pass `repo_url` and not `repo_path`?